hs_modelinstance/static/js/model_instance.js

```diff
@@ -57,7 +149,35 @@ function show_model_details(e) {
         },
         error: function (data) {
             console.log('There was an error with model instance GET.')
+        },
+        complete: function(data){
+
+            // set the modelprogram id of the selected value
+            set_hidden_model_name(shortid);
+
+            // enable/disable the save button if the value has been changed to something new
+            if (shortid != mp_old_id)
+                $('#id-executedby').find('.btn-primary').show();
+            else
+                $('#id-executedby').find('.btn-primary').hide();
         }
     });
 
 }
+
+/*
+    Sets the ModelProgram shortid to the hidden model_name field.  This field is submitted to django and used to set
+    database fields ModelName and ModelForeignKey.
+ */
+function set_hidden_model_name(id){
+
+    var model_name = document.getElementById('id_model_name');
+    model_name.value = id;
+}
+
```

Comment:
This function doesn'd do much. I would just use document.getElementById('id_model_name').value = id where necessary.